File: API/_archive/gRPC/golang/01_inbuilt_basic.go
// BASIC EXAMPLE: Go Inbuilt net/rpc
// Demonstrates the native binary RPC standard library in Go (the precursor to gRPC).
package main

import (
	"log"
	"net"
	"net/rpc"
)

type UserService struct{}
type GetUserArgs struct{ ID string }
type User struct{ Name string }

// Method must have this exact signature to be exported via RPC
func (s *UserService) GetUser(args *GetUserArgs, reply *User) error {
	reply.Name = "Alice"
	return nil
}

func main() {
	rpc.Register(new(UserService))
	listener, err := net.Listen("tcp", ":1234")
	if err != nil {
		log.Fatal("Listen error:", err)
	}
	
	log.Println("Go Standard RPC Server running on :1234")
	rpc.Accept(listener)
}

File: API/_archive/gRPC/python/01_inbuilt_basic.py
# BASIC EXAMPLE: Python Inbuilt xmlrpc.server
# Demonstrates the standard library precursor to gRPC (XML-RPC).
from xmlrpc.server import SimpleXMLRPCServer

def get_user(id_str):
    if id_str == "1":
        return {"id": "1", "name": "Alice"}
    return {}

if __name__ == "__main__":
    server = SimpleXMLRPCServer(("localhost", 8000))
    print("XML-RPC Listening on port 8000...")
    server.register_function(get_user, "get_user")
    server.serve_forever()

    # To test as a client:
    # import xmlrpc.client
    # proxy = xmlrpc.client.ServerProxy("http://localhost:8000/")
    # print(proxy.get_user("1"))
